hs-bindgen/src-internal/HsBindgen/Frontend/OmittedDecls.hs

```diff
@@ -0,0 +1,37 @@
+module HsBindgen.Frontend.OmittedDecls (
+    -- * Type
+    OmittedDecls(..)
+    -- * API
+  , empty
+  , insert
+  , lookup
+  ) where
+
+import Data.Map.Strict qualified as Map
+import Prelude hiding (lookup)
+
+import Clang.Paths (SourcePath)
+import HsBindgen.Imports
+import HsBindgen.Language.C.Name (CName, Namespace)
+
+{-------------------------------------------------------------------------------
+  Type
+-------------------------------------------------------------------------------}
+
+newtype OmittedDecls = OmittedDecls {
```

Comment:
Let's rename this to `NonSelectedDecls`.

Reply:
We should add a comment explaining that these are the declarations that were not selected during parsing.

Add further comment: It is important to keep track of these non-selected declarations because they can be given external bindings. We do _not_ support external bindings for _anonymous_ non-selected declarations; _if_ you want to provide an external binding for some local type, for example

```c
struct rect {
  struct { int x; int y; } bottomleft;
  struct { int x; int y; } topright;
}
```

then you need to make sure that you _traverse_ `rect`, so that the `RenameAnon` pass can do its work.